vscode/extension.js

```diff
@@ -94,29 +90,43 @@ var KiteOutgoing = class {
   // Sends an event to Kite.
   sendEvent (action) {
     let event = this.buildKiteEvent(action);
-    let msg = JSON.stringify(event);
-
-    this.outgoingSocket.send(msg, 0, msg.length, KiteOutgoing.PORT, KiteOutgoing.HOST);
+    this.httpRoundTrip('/clientapi/editor/event', event);
   }
 
   // Sends an error to Kite.
-  sendError (data) {
+  sendError (msg) {
     const editor = vscode.window.activeTextEditor;
 
     if (!editor) {
       return;
     }
 
-    let event = {
+    this.httpRoundTrip('/clientapi/editor/error', {
       'source': SOURCE,
-      'action': "error",
       'filename': editor.document.fileName,
-      'text': JSON.stringify(data),
-      'pluginId': PLUGIN_ID,
+      'text': msg,
+    });
+  }
+
+  // httpRoundTrip - performs a POST request and discards the result
+  httpRoundTrip (endpoint, obj) {
+    var payload = JSON.stringify(obj);
+    if (payload.length > MAX_PAYLOAD_SIZE) {
+      console.log("unable to send message because length exceeded limit");
+      reset();
+      return;
+    }
+
+    var options = {
+      host: '127.0.0.1',
+      port: '46624',
+      path: endpoint,
+      method: 'POST',
     };
 
-    let msg = JSON.stringify(event);
-    this.outgoingSocket.send(msg, 0, msg.length, KiteOutgoing.PORT, KiteOutgoing.HOST);
+    var req = http.request(options);
+    req.write(payload);
+    req.end();
```

Comment:
hmm i think you are correct http://blog.mixu.net/2011/02/01/understanding-the-node-js-event-loop/